be/src/exec/aggregate/distinct_blocking_node.cpp

```diff
@@ -135,7 +135,8 @@ pipeline::OpFactories DistinctBlockingNode::_decompose_to_pipeline(pipeline::OpF
     using namespace pipeline;
 
     auto workgroup = context->fragment_context()->workgroup();
-    auto executor = std::make_shared<spill::IOTaskExecutor>(ExecEnv::GetInstance()->scan_executor(), workgroup);
+    auto executor = std::make_shared<spill::IOTaskExecutor>(
```

Comment:
Now the lifecycle of the executor is aligned with the query ctx. However, in some cases, the query ctx is already destructed, but the task is still in the task queue, and the executor is already destructed. Accessing the callback at this point will result in use-after-tree.

Reply:
> Now the lifecycle of the executor is aligned with the query ctx. However, in some cases, the query ctx is already destructed, but the task is still in the task queue, and the executor is already destructed. Accessing the callback at this point will result in use-after-tree.

will fix it in next PR